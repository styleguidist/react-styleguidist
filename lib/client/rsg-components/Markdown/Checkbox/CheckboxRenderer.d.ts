import React from 'react';
import { JssInjectedProps } from 'rsg-components/Styled';
export declare const CheckboxRenderer: React.FunctionComponent<JssInjectedProps>;
declare const _default: React.ComponentType<Pick<JssInjectedProps, never>>;
export default _default;
