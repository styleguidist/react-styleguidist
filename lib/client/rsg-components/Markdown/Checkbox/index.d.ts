export { default } from 'rsg-components/Markdown/Checkbox/CheckboxRenderer';
