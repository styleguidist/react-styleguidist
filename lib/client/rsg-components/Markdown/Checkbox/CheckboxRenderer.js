import "core-js/modules/es.array.index-of";
import "core-js/modules/es.object.assign";
import "core-js/modules/es.object.keys";

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';

var styles = function styles() {
  return {
    input: {
      isolate: false,
      display: 'inline-block',
      verticalAlign: 'middle'
    }
  };
};

export var CheckboxRenderer = function CheckboxRenderer(_ref) {
  var classes = _ref.classes,
      rest = _objectWithoutPropertiesLoose(_ref, ["classes"]);

  return /*#__PURE__*/React.createElement("input", _extends({}, rest, {
    type: "checkbox",
    className: classes.input
  }));
};
CheckboxRenderer.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired
};
export default Styled(styles)(CheckboxRenderer);