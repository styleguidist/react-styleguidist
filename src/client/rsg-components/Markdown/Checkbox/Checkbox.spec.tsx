import React from 'react';
import renderer from 'react-test-renderer';

import Checkbox from './index';

describe('Markdown Checkbox', () => {
	it('should render a checkbox input', () => {
		const actual = renderer.create(<Checkbox />);

		expect(actual.toJSON()).toMatchSnapshot();
	});
});
