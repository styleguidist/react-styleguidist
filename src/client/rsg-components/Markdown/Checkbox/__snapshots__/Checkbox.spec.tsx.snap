// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown Checkbox should render a checkbox input 1`] = `
<input
  className="rsg--input-0"
  type="checkbox"
/>
`;
